lib/SourceMapDevToolPlugin.js

```diff
@@ -273,10 +273,9 @@ class SourceMapDevToolPlugin {
 							if (currentSourceMappingURLComment !== false) {
 								assets[file] = compilation.assets[file] = new ConcatSource(
 									new RawSource(source),
-									currentSourceMappingURLComment.replace(
-										/\[url\]/g,
-										sourceMapUrl
-									)
+									currentSourceMappingURLComment
```

Comment:
There is a util function for processing these placeholders -> `compilation.getPath(...)` (see also for `sourceMapFile` above)

Pass `currentSourceMappingURLComment` through `compilation.getPath` to get all default placeholders and the ability to add placeholders via plugins.

Support for `[url]` should than be moved to `TemplatedPathPlugin` which is responsible for the default placeholders.